### ISSUE TITLE
support azure file snapshot

### ISSUE THREAD

**Issue by andyzhangx** [heptio/ark#1046]
Currently azure cloud provide only supports azure disk snapshot, and [azure file plugin](https://kubernetes.io/docs/concepts/storage/volumes/#azurefile) is also a common storage layer which could provide multiple read/write.
Azure file snapshot doc:
https://docs.microsoft.com/en-us/azure/storage/files/storage-snapshots-files

@rosskukulinski

**Comment by ncdc** [comment]
@andyzhangx what would the mechanics be around restores? With Azure managed disks, for example, we create a new managed disk from a snapshot and associate that with the Kubernetes PersistentVolume. Could we do something similar with Azure files - can we create a new "share" from a snapshot and associate that with a PV? Or how would you envision this working? Thanks!

**Comment by andyzhangx** [comment]
Yes, azure file pvc is like azure disk pvc. It's using different protocol and support multiple readwrite.
And its apiVersion requires >= 2017-04-17, see RESET api here:
https://docs.microsoft.com/en-us/rest/api/storageservices/snapshot-share

**Comment by moebius87** [comment]
Hi @ncdc , is there any good news about this issue? Thank you!

**Comment by skriss** [comment]
@moebius87 nobody has picked this up to work on it, but it sounds like there's a good chance that a plugin could be developed for this. I'm happy to do a little R&D and provide some direction if someone wants to work on it!

**Comment by andyzhangx** [comment]
Update: we are already implementing azure file snapshot functionality in https://github.com/kubernetes-sigs/azurefile-csi-driver, currently azure file does not provide a restore sdk for this functionality, and we are expecting azure file team to provide a v2 sdk to cover this functionality. I think we could wait until that restore function is released, stay tuned.
